File: 807-custom-sort-string/custom-sort-string.java
class Solution {
    public String customSortString(String order, String s) {
        Map<Character, Integer> mapForFreq = new HashMap<>();
        String result = "";
        for(Character ch: s.toCharArray()){
            mapForFreq.put(ch, mapForFreq.getOrDefault(ch,0)+1);
        }

        for(Character ch: order.toCharArray()){
            if(mapForFreq.containsKey(ch)){
                int freq= mapForFreq.get(ch);
                while(freq!=0){
                    result += ch;
                    freq--;
                } 
                mapForFreq.remove(ch);
            }
        }
        
        for(Character key:mapForFreq.keySet()){
            int freq = mapForFreq.get(key);
            while(freq!=0){
                    result += key;
                    freq--;
            }
        }

        return result;

    }
}

File: 807-custom-sort-string/custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        unordered_map<char, int> mapOfS;
        string result;
        
        for(auto charOfS: s) mapOfS[charOfS]++;

        for(auto charOfOrder: order){
            if(mapOfS.find(charOfOrder) != mapOfS.end()){
                string s(mapOfS[charOfOrder], charOfOrder);
                result += s;
                mapOfS.erase(charOfOrder);
            }
        } 

        for(auto entry: mapOfS){
            string s(entry.second, entry.first);
            result+=s;
        }

        return result;
    }
};